test/test_evaluator.cpp

```diff
@@ -558,4 +558,72 @@ TEST_F(ExprEvalTest, Permute) {
   EXPECT_THAT(out_tensor.strides(), ElementsAre(128, 1, 32, 8));
 }
 
+TEST_F(ExprEvalTest, ReshapePermuteReshape) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* in =
+      TensorViewBuilder().shape({-1, 6}).dtype(DataType::Float).build();
+  fusion.addInput(in);
+  TensorView* out = reshape(
+      in, {size(in, 0), IrBuilder::create<Val>(2), IrBuilder::create<Val>(3)});
+  out = permute(out, {1, 2, 0});
+  out = reshape(out, {IrBuilder::create<Val>(6), size(out, 2)});
+  fusion.addOutput(out);
+
+  at::Tensor in_tensor = at::rand({72}).cuda().as_strided({9, 6}, {8, 1});
+
+  ExpressionEvaluator evaluator;
+  evaluator.bind(in, in_tensor);
+  at::Tensor out_tensor = evaluator.evaluate(out).as<at::Tensor>();
+
+  EXPECT_EQ(in_tensor.data_ptr(), out_tensor.data_ptr());
+  EXPECT_THAT(out_tensor.sizes(), ElementsAre(6, 9));
+  EXPECT_THAT(out_tensor.strides(), ElementsAre(1, 8));
+}
+
+TEST_F(ExprEvalTest, Reshape_ForwardBroadcast) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  TensorView* in =
+      TensorViewBuilder().shape({-1, 6}).dtype(DataType::Float).build();
```

Comment:
Should we set the first dim to expanded?
https://github.com/NVIDIA/Fuser/blob/5a22d44ee8a4d8a8a6dd2ce63016552be61fbc43/csrc/ir/interface_nodes.h#L630

Reply:
TIL! This made it possible for me to test expanded broadcasts better and actually find a bug.